vignettes/tcplfit2-vignette.Rmd

```diff
@@ -362,3 +362,148 @@ The output table resulting from the previous code chunk is the same format as th
+```{r example 5 lower}
+# load example data
+spid <- unique(mc3$spid)[94]
+temp <- mc3[is.element(mc3$spid,spid),]
+
+# The data file has stored concentration in log10 form, fix it 
+conc <- 10**temp$logc # back-transforming concentrations on log10 scale
+resp <- temp$resp
+
+# modify the data for demonstration purposes 
+conc2 <- conc[conc>0.41]
+resp2 <- resp[which(conc>0.41)]
+
+# pull out all of the chemical identifiers and the name of the assay
+dtxsid <- temp[1,"dtxsid"]
+casrn <- temp[1,"casrn"]
+name <- temp[1,"name"]
+assay <- temp[1,"assay"]
+
+# create the row object
+row <- list(conc = conc2, resp = resp2, bmed = 0, cutoff = cutoff, onesd = onesd,
+            assay=assay, dtxsid=dtxsid,casrn=casrn,name=name)
+
+# run the concentration-response modeling for a single sample
+res_low <- concRespCore(row,fitmodels = c("cnst", "hill", "gnls", "poly1", "poly2", 
+                                          "pow", "exp2", "exp3", "exp4", "exp5"), 
+                        conthits = T, aicc = F, bidirectional=F)
+
+concRespPlot(res_low,ymin=-10,ymax=100)
+```
+
+```{r example 5 lower-res, echo=FALSE}
+# function results
+res_low['Min. Conc.'] <- min(conc2)
+res_low['Name'] <- name
+res_low[1, c("Min. Conc.", "bmd", "bmdl", "bmdu")] <- round(res_low[1, c("Min. Conc.", "bmd", "bmdl", "bmdu")], 3)
+DT::datatable(res_low[1, c("Name","Min. Conc.", "bmd", "bmdl", "bmdu")],rownames = FALSE)
+```
+
+The lowest concentration in the data is 0.6 but the estimated BMD from the function is 0.302, it is lower than the minimum concentration tested. One can allow BMD to be lower than the minimum concentration tested while restrict it to be no lower than a percentage of that by using the argument `bmd_low_bnd`. For example, if one sets this argument to 0.8, the BMD is not allowed to be lower than 80% of the lowest concentration in the data (i.e., 0.8*0.6 = 0.48). If the BMD is lower than the minimum concentration but not lower than this lower bound, the function will returned it as-is. However, in this example, 0.302 is still lower than 0.48, so the function will force BMD to be on the lower bound and shift the original BMD boundaries by the difference between the BMD estimate and the lower bound. One might set `bmd_low_bnd` to 1 to avoid BMD to be lower than the minimum concentrate in the data. 
+```{r example 5 lower-demo}
+# using the argument to set a lower bound for BMD
+res_low2 <- concRespCore(row,fitmodels = c("cnst", "hill", "gnls", "poly1", "poly2", 
+                                           "pow", "exp2", "exp3", "exp4", "exp5"), 
+                         conthits = T, aicc = F, bidirectional=F, bmd_low_bnd = 0.8)
+```
+
+```{r example 5 new lower-res, echo=FALSE}
```

Comment:
May want to consider putting the original results next to the censored results to demonstrate quantitatively the difference.  Please make sure to designate appropriately if you incorporate this.